File: Chat/Chat/wrapper/java/JChatSDK/src/com/ringid/imsdk/GroupInformation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.ringid.imsdk;

/**
 *
 * @author towhid
 */
public class GroupInformation {
    private long groupId;
    private String registerServerIp;
    private int registerServerPort;
    private int bindingPort;
    private long lastActivityTime;

    public GroupInformation(long groupId, String registerServerIp, int registerServerPort, int bindingPort, long lastActivityTime) {
        this.groupId = groupId;
        this.registerServerIp = registerServerIp;
        this.registerServerPort = registerServerPort;
        this.bindingPort = bindingPort;
        this.lastActivityTime = lastActivityTime;
    }

    /**
     * @return the registerServerIp
     */
    public String getRegisterServerIp() {
        return registerServerIp;
    }

    /**
     * @param registerServerIp the registerServerIp to set
     */
    public void setRegisterServerIp(String registerServerIp) {
        this.registerServerIp = registerServerIp;
    }

    /**
     * @return the registerServerPort
     */
    public int getRegisterServerPort() {
        return registerServerPort;
    }

    /**
     * @param registerServerPort the registerServerPort to set
     */
    public void setRegisterServerPort(int registerServerPort) {
        this.registerServerPort = registerServerPort;
    }

    /**
     * @return the bindingPort
     */
    public int getBindingPort() {
        return bindingPort;
    }

    /**
     * @param bindingPort the bindingPort to set
     */
    public void setBindingPort(int bindingPort) {
        this.bindingPort = bindingPort;
    }

    /**
     * @return the groupId
     */
    public long getGroupId() {
        return groupId;
    }

    /**
     * @param groupId the groupId to set
     */
    public void setGroupId(long groupId) {
        this.groupId = groupId;
    }

    /**
     * @return the lastActivityTime
     */
    public long getLastActivityTime() {
        return lastActivityTime;
    }

    /**
     * @param lastActivityTime the lastActivityTime to set
     */
    public void setLastActivityTime(long lastActivityTime) {
        this.lastActivityTime = lastActivityTime;
    }
    
    
}

File: Chat/Chat/src/GroupInformation.cpp
/*
* File:   GroupInformation.h
* Author: noman
*
* Created on May 16, 2016, 1:12 PM
*/

#include "GroupInformation.h"

namespace imsdk{
	GroupInformation::GroupInformation(GroupIdentity groupId, std::string serverIp, int serverPort, int chatBindingPort, long long lastActivityTime) :groupId(groupId), registerServerIp(serverIp), registerServerPort(serverPort), bindingPort(chatBindingPort), lastActivityTime(lastActivityTime)
	{
	}


	GroupInformation::~GroupInformation()
	{
	}

	std::string GroupInformation::getRegisterServerIp()
	{
		return registerServerIp;
	}

	void GroupInformation::setRegisterServerIp(std::string registerServerIp)
	{
		this->registerServerIp = registerServerIp;
	}

	int GroupInformation::getRegisterServerPort()
	{
		return registerServerPort;
	}

	void GroupInformation::setRegisterServerPort(int registerServerPort)
	{
		this->registerServerPort = registerServerPort;
	}

	int GroupInformation::getBindingPort()
	{
		return bindingPort;
	}

	void GroupInformation::setBindingPort(int bindingPort)
	{
		this->bindingPort = bindingPort;
	}

        void GroupInformation::setLastActivityTime(long long lastActivityTime) {
            this->lastActivityTime = lastActivityTime;
        }

        long long GroupInformation::getLastActivityTime() {
            return lastActivityTime;
        }

        void GroupInformation::setGroupId(GroupIdentity groupId) {
            this->groupId = groupId;
        }

        GroupIdentity GroupInformation::getGroupId() {
            return groupId;
        }
}